File: 0198-house-robber/0198-house-robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        rob1 = rob2 = 0
        # [rob1, rob2, n, n+1, ...]
        for num in nums:
            # rob1, rob2 = rob2, max(num + rob1, rob2) # assign both 
            temp = max(num + rob1, rob2)
            rob1 = rob2
            rob2 = temp
            
        return rob2

File: 0198-house-robber/0198-house-robber.cpp
class Solution {
private:

public:
    int rob(vector<int>& nums) {
        int rob1=0;
        int rob2=nums[0];
        int temp=0;
        for(int i=1;i<nums.size();i++){
            temp=max(rob2,rob1+nums[i]);
            rob1=rob2;
            rob2=temp;
        }
        return rob2;
    }
};